src/core/libmaven/peakdetector.cpp

```diff
@@ -992,6 +992,7 @@ void PeakDetector::linkParentIsotopeRange(PeakGroup& parentGroup,
                 child->deletePeak(sample);
         }
 
+        delete_all(eics);
```

Comment:
Was this the reason for crash while detecting peaks, reported by testing team?

Reply:
Yes. Entire EICs for each peak-group (and its children) were being created but not being deallocated during this step. And each EIC is a large object - this was a very bad memory leak.